core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/instance/InstanceGroup.java

```diff
@@ -186,7 +186,7 @@ public InstanceGroup replaceInstanceMetadata(Set<InstanceMetaData> instanceMetaD
 
     public Set<InstanceMetaData> getUnattachedInstanceMetaDataSet() {
         return instanceMetaData.stream()
-                .filter(metaData -> metaData.getInstanceStatus() == InstanceStatus.CREATED)
+                .filter(metaData -> metaData.getInstanceStatus() == InstanceStatus.CREATED || metaData.getInstanceStatus() == InstanceStatus.DECOMMISSIONED)
```

Comment:
As part of a downscale, the first step is to decommission instances via CM - and the state for these instances is then set to DECOMMISSIONED.

After this, the instances are stopped via the cloud-provider, and the state is set to STOPPED.

If there's a failure after the first step - the instances will end up in a DECOMMISSIONED state. Going forward, the expectation is that such instances will be picked up in either 1) subsequent downscales to retry a stop, or 2) subsequent upscales to avoid starting nodes unnecessary.

In the meantime, I think this change makes such nodes candidates for some downscale operation (not sure which downscale operation though. I know downscale by N nodes today ends up preferring instances which are in STOPPED state, if they exist). This likely adds instances in the DECOMMISSIONED state to the same list?
(Only place getUnattachedInstanceMetaDataSet is used today ... not sure where it'll be used in the future)

Assuming the above is correct - in terms of this serving as a node selection criteria (and that a downscale can't start in the middle of a downscale via STOP operation), I don't think this causes issues with stop-start scaling - since all it's doing is enhancing the collection set for which nodes to downscale (better to remove DECOMMISSIONED, STOPPED compared to HEALTYH)

Reply:
I will remove this.